src/transform/src/fusion/map.rs

```diff
@@ -52,11 +43,14 @@ impl Map {
                 inner_scalars.append(scalars);
                 mem::swap(scalars, inner_scalars);
                 **input = inner_input.take_dangerous();
+                inputs.take_first();
```

Comment:
ooops, this is not even correct. It should be `inputs.first_mut().inputs.take_first()`